--- origsrc/gob2-2.0.15/src/Makefile.am	2007-03-09 11:46:14.000000000 -0600
+++ src/gob2-2.0.15/src/Makefile.am	2007-11-28 07:14:15.310940900 -0600
@@ -61,15 +61,15 @@
 # 	test -f y.tab.h && mv -f y.tab.h parse.h
 
 test: test-object.c test-object.h
-	$(CC) -g -O -W -Wall -Wunused -Wuninitialized -Wdeclaration-after-statement -Wshadow -Wwrite-strings -Wunreachable-code -Wbad-function-cast `pkg-config --cflags gobject-2.0 gtk+-2.0` -c test-object.c
+	$(CC) $(CFLAGS) -W -Wall -Wunused -Wuninitialized -Wdeclaration-after-statement -Wshadow -Wwrite-strings -Wunreachable-code -Wbad-function-cast `pkg-config --cflags gobject-2.0 gtk+-2.0` -c test-object.c
 
-test-object.c test-object.h: test.gob gob2
-	./gob2 test.gob
+test-object.c test-object.h: test.gob gob2$(EXEEXT)
+	./gob2 $(srcdir)/test.gob
 
 # can't depend on the header file as that would break the above rule
 # since it's just for testing, who cares
 testcpp: test-object.cc
-	g++ -g -O -W -Wall -Wunused -Wuninitialized  -Wwrite-strings -Wunreachable-code `pkg-config --cflags gobject-2.0 gtk+-2.0` -c test-object.cc
+	g++ $(CFLAGS) -W -Wall -Wunused -Wuninitialized  -Wwrite-strings -Wunreachable-code `pkg-config --cflags gobject-2.0 gtk+-2.0` -c test-object.cc
 
-test-object.cc: test.gob gob2
-	./gob2 --for-cpp test.gob
+test-object.cc: test.gob gob2$(EXEEXT)
+	./gob2 --for-cpp $(srcdir)/test.gob
